x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -888,3 +912,140 @@ func (s *KeeperTestSuite) TestGetUnbondingICAMessages() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestConsolidateUnbondingMessages_Success() {
+	batchSize := 4
+	totalUnbondAmount := int64(1501)
+	excessUnbondAmount := int64(101)
+
+	validatorMetadata := []struct {
+		address                    string
+		initialUnbondAmount        int64
+		remainingDelegation        int64
+		expectedDelegationIncrease int64
+	}{
+		// Total Remaining Portion: 1000 + 500 + 250 + 250 = 2000
+		// Excess Portion = Remaining Delegation / Total Remaining Portion
+
+		// Excess Portion: 1000 / 2000 = 50%
+		// Delegation Increase: 50% * 101 = 50
+		{address: "val1", initialUnbondAmount: 500, remainingDelegation: 1000, expectedDelegationIncrease: 50},
```

Comment:
Looks like we're testing with less than 32 vals. Do you think that's sufficient coverage if the batchSize is also low?

Reply:
Yeah since batchSIze is passed in as a param, I think it's functionally the same

Reply:
I'm not confident the coverage is equivalent (I think there could be cases we'd hit with more vals, e.g. there could be much more spillover that needs to be distributed across the batchSize set) but I'm not very confident in them, couldn't point you to a specific case. As long as you did some checks with the unbonding simulator tool spreadsheet as well, I think I feel good about this. Sounds like you did some test with the unbonding simulator?

Reply:
I would just think about it in terms of the different permutations of outcomes, which is generally agnostic to the batch size. 

From that framing, the only cases I could think of were:
* We are able to successfully re-distribute the excess  [[covered here](https://github.com/Stride-Labs/stride/pull/1018/files#diff-42324da485a3df105dabdb479a72a8b7e82535250b0742a415e1c590081913ffR916)]
* There is not enough existing stake to redistribute the access [[tested here](https://github.com/Stride-Labs/stride/pull/1018/files#diff-42324da485a3df105dabdb479a72a8b7e82535250b0742a415e1c590081913ffR1023)]
* There is no excess [covered by the `TestUnbondFromHostZone` cases]

Very open to adding more tests if you have a case in mind that doesn't fall into one of these!

Reply:
From riley on slack: 
> Yeah that makes sense. And I can't think of other cases. If I think of any, will follow up.

Marking this resolved!